tabular/src/autogluon/tabular/models/catboost/catboost_model.py

```diff
@@ -53,9 +53,60 @@ def _preprocess_nonadaptive(self, X, **kwargs):
                     X[category] = X[category].cat.add_categories('__NaN__').fillna('__NaN__')
         return X
 
+    #TODO: could be useful if we could still exploit the early stopping feature during incremental training
+    #incremental training appears not compatible with early stopping for now
+    def _set_early_stopped(self, progress, iterations, early_stopping_rounds):
+        if progress:
+            self._early_stopped = iterations > (self.model.get_best_iteration() + early_stopping_rounds) \
+            if early_stopping_rounds else False
+        else:
+            self._early_stopped = True
+
+    def _merge_prev_model(self, init_model, metric_name, X_val=None, y_val=None):
+        progress = True #test if adding additional trees help with accuracy
+        if init_model is not None:
+            init_model_best_score = init_model.get_best_score()['validation'][metric_name]
+            final_model_best_score = self.model.get_best_score()['validation'][metric_name]
+            if self.stopping_metric._optimum > final_model_best_score:
+                if final_model_best_score > init_model_best_score:
+                    best_iteration = init_model.tree_count_ + self.model.get_best_iteration()
```

Comment:
code duplicate, in this are just determine if prev or current is better, then outside of if statements set best_iteration and progress accordingly.